database/seeds/RegistrationsSeeder.php

```diff
@@ -37,9 +37,120 @@ public function run()
         // create a registration with a bundle
         $bundle = factory(App\Bundle::class)->create();
         factory(App\Registration::class)->create()->bundles()->save($bundle);
-    }
+
+        // centre without SK rules for fam1 & fam4 that is not in same area as families 3, 5 and 6
+        $testCentreFam1And4 = App\Centre::find(7);
+        $testCentreFam1And4->sponsor_id = App\Sponsor::find(5)->id;
```

Comment:
I think you should get the sponsor id from the centre so that they are consistent with the data

        `$testCentreFam1And4->sponsor->id;`

Reply:
the reason I'm doing a re-assignment (a centre is attached to a sponsor by default it seems) is to make sure that each of these centres are in a different area (that is one of the requests on the ticket). if I just the centre's sponsor id we run the risk of having 2 or more centres in the same areas because this is randomly assigned. at least that's my understanding so correct me if I'm wrong. 🧇

Reply:
It might be better to set up the centres in their seeds instead?

Reply:
I don't think you should be altering them here.

Reply:
Especially since you don't save the centre after. It will make the data inconsistent.

Reply:
so then, not to use the existing centres but create new ones? again, is it right that creating a new centre assigns it to a sponsor/area _randomly_? if so, there's a chance that some of these test families will be in the same area and that's what we're trying to avoid here.

Reply:
Since a centre requires a sponsor to save itself, the factory 
- creates one if there are none
- randomly picks one of the existing ones if there's one or more

So, when using a factory you need to reassign the sponsor if it's not the one you want.
Then save the centre so the seed is persisted in DB

If you need a centre for a specific test, and one centre does not seem to be precisely appropriate, probably best to make a new one, as we can't garuantee that it wont cause strange issues if other seeds use the same centre.

Reply:
well, from all this I take that:
1) if I don't want any funny behaviour of the seeds these test families should have new centres and not use existing ones. So better to be defensive here. Question is: should I or not? I understand the argument but what would **you** do? I'd want to create new centres for these test families even though existing ones seem appropriate but there's an unquantifiable risk of side effects from other seeds using those centres. 
2) might be better to set the centres in the seeds - readability? or is there any other reason?

 If you want me to make a decision here, I will but it seems I made the wrong or sub-optimal decisions so far. Otherwise we wouldn't be having this discussion 😄

Reply:
I think you are being a bit hard on yourself. My opinion. If the existing seeded centres are not consistent (i.e. random on generation) or not suitably configured, make new ones.

If the existing ones work for you, use them as they are or alter them when they get generated in their own seeder, rather than here.

Reply:
I moved the centres inside the seeds where these are created anew (with their own sponsor id) and the seeds are now in a private function. It's a cleaner look from my point of view,separating concerns. hope it's better now.